rfc/accepted/0021-command-driven-connectors.md

```diff
@@ -0,0 +1,122 @@
+# Command driven connectors
+
+- Feature Name: `command_driven_connectors`
+- Start Date: 2021-10-07
+- Tremor Issue: [tremor-rs/tremor-runtime#0000](https://github.com/tremor-rs/tremor-runtime/issues/0000)
+- RFC PR: [tremor-rs/tremor-rfcs#0000](https://github.com/tremor-rs/tremor-rfcs/pull/0000)
+
+## Summary
+[summary]: #summary
+
+This RFC adds support for connectors that don't do anything by themselves, but instead are driven by commands sent to them.
+
+## Motivation
+[motivation]: #motivation
+
+There are many connectors, where, especially on the read side, there's no obvious "default" way to read data. For example object stores usually do not provide an API to stream changes (and if they did, the users might still want to read files in a different fashion). With command driven connectors, users will be able to read data that they need based on the commands they send to the connector.
+
+## Guide-level Explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Command-driven connectors define at least two ports - one for data and one for commands.
+As an example, let's look at a connector that reads files:
+
+```tremor
+define flow main
+flow
+  define connector file_connector from file
+    with 
+      codec="string",
+      config = {"command_driven": true}
+  end;
+  
+  define connector file_list from file
+    with 
+    codec = "json",
+    config = {
+      "path": "in.json",
+      "mode": "read", 
+    },
+  end;
+  
+  create pipeline main
+  pipeline
+    select { "command": "read", "path": event.path } from in into out;
+  end;
+  
+  create connector file_connector from file_connector;
+  create connector file_list from file_list;
+  
+  connect /connector/file_list/out to /pipeline/main/in;
+  
+  # This is the magic - we send the commands here, note the "commands" port
+  connect /pipeline/main/out to /connector/file_connector/commands;

```

Comment:
`control` might be a better name for the port, the usual distinction in systems like this is between `data plane` and `control plane`